packages/hardhat-core/src/builtin-tasks/flatten.ts

```diff
@@ -70,11 +71,12 @@ subtask(
   "Returns all contracts and their dependencies flattened"
 )
   .addOptionalParam("files", undefined, undefined, types.any)
+  .addOptionalParam("unifyABIEncoderV2", undefined, false, types.boolean)
   .addOptionalParam("removeLicenses", undefined, false, types.boolean)
   .addOptionalParam("license", undefined, undefined, types.string)
   .setAction(
     async (
-      { files, removeLicenses = false, license }: FlattenInput,
+      { files, unifyABIEncoderV2 = false, removeLicenses = false, license }: FlattenInput,
```

Comment:
This option seems very ad hoc. I understand that this is what was required in the issue, but it doesn't feel right.

Maybe it could be called `unifyPragmas`, even if it only does this for the ABI encoder v2 one. If it happens that there's another pragma that needs this, we can add it in the future.

@alcuadrado wdyt?

Reply:
See also: https://github.com/nomiclabs/hardhat/issues/1050#issuecomment-838308098

Reply:
@fvictorio I have added it cause I forgot to add it when I made the PR. 

From Remix, you get the following errors when you flattened contracts with _ABIEncoder V2_. I would like to keep that flag.
<img width="298" alt="Screen Shot 2021-05-11 at 08 32 32" src="https://user-images.githubusercontent.com/7549152/117808581-715e1700-b233-11eb-9867-0b95d5e0ea1e.png">

About the naming, `unifyPragmas` can be confused because you can still have different _pragma solidity_ versions. Maybe `unifyExperimentalPragmas` works can't predict if we will have more experimental pragmas in the future.